Extensions/xkit_preferences.js

```diff
@@ -2444,6 +2463,9 @@ XKit.extensions.xkit_preferences = new Object({
 
 	destroy: function() {
 		$("#xkit_button").remove();
+		if (this.xkit_button_observer) {
+			this.xkit_button_observer.disconnect();
```

Comment:
we should make sure to disconnect the observer *before* removing the button just so we don't accidentally trigger the observer when removing the button ourselves.